pkg/actuators/machine/instances_test.go

```diff
@@ -151,29 +151,31 @@ func TestGetServiceInstanceID(t *testing.T) {
 	cases := []struct {
 		name                           string
 		instanceName                   string
-		serviceInstance                powervsproviderv1.PowerVSResourceReference
+		serviceInstance                machinev1.PowerVSResource
 		getCloudServiceInstancesValues []bluemixmodels.ServiceInstanceV2
 		getCloudServiceInstancesError  error
 		expectedError                  string
 	}{
 		{
 			name: "With ServiceInstanceID",
-			serviceInstance: powervsproviderv1.PowerVSResourceReference{
-				ID: core.StringPtr(instanceID),
+			serviceInstance: machinev1.PowerVSResource{
+				Type: machinev1.PowerVSResourceTypeID,
+				ID:   core.StringPtr(instanceID),
 			},
 		},
 		{
 			name:         "With valid ServiceInstanceName",
 			instanceName: instanceName,
-			serviceInstance: powervsproviderv1.PowerVSResourceReference{
+			serviceInstance: machinev1.PowerVSResource{
+				Type: machinev1.PowerVSResourceTypeName,
 				Name: core.StringPtr(instanceName),
 				ID:   core.StringPtr(instanceID),
 			},
 			getCloudServiceInstancesValues: []bluemixmodels.ServiceInstanceV2{
 				{
 					ServiceInstance: bluemixmodels.ServiceInstance{
 						MetadataType: &bluemixmodels.MetadataType{
-							ID: instanceID,
+							Guid: instanceID,
```

Comment:
why this flip? `ID` => `Guid`?

Reply:
Guid is the one which actually containst the serviceInstance id